finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
```

Comment:
Yes, making server bridge `ChannelInboundHandler` and moving code from `channelInit` to `channelRegistered`. 
Could you help me understand why if I change `ServerBridge` like this:
```scala
  def initChannel(ch: Channel): Unit = {
    val sslHandler = ch.pipeline().get(classOf[SslHandler])
    if (sslHandler != null) {
      sslHandler.handshakeFuture().addListener(new GenericFutureListener[NettyFuture[Channel]] {
        override def operationComplete(future: NettyFuture[Channel]): Unit = {
          if (future.isSuccess) {
            val transport: Transport[In, Out] = transportFac(ch)
            serveTransport(transport)
          }
        }
      })
    } else {
      val transport: Transport[In, Out] = transportFac(ch)
      serveTransport(transport)
    }
  }
```
handshake does not happen?